validator_client/src/lib.rs

```diff
@@ -342,6 +342,9 @@ impl<T: EthSpec> ProductionValidatorClient<T> {
                 log.clone(),
             ));
 
+        // Ensure all validators are registered in doppelganger protection.
+        validator_store.register_all_in_doppelganger_protection_if_enabled()?;
```

Comment:
Since this now happens before `wait_for_genesis` I think this may cause the VC to fail to start pre-genesis because of this piece of code:

https://github.com/paulhauner/lighthouse/blob/doppel-crit/validator_client/src/doppelganger_service.rs#L373-L376

But I think updating `slot_clock.now()` to `slot_clock.now_or_genesis()` should fix it

Reply:
Nice catch!